test-app/tests/integration/components/multiselect-test.gts

```diff
@@ -0,0 +1,1513 @@
+
+        <:default as |multiselect|>
+          <multiselect.Option>{{multiselect.option}}</multiselect.Option>
+        </:default>
+      </Multiselect>
+    </template>);
+
+    assert.verifySteps([]);
+
+    await fillIn('[data-multiselect]', 'blue');
+
+    assert.dom('[role="option"]').exists({ count: 1 });
+
+    await triggerKeyEvent('[data-multiselect]', 'keydown', 'Enter');
+
+    assert.verifySteps(['handleChange']);
+
+    // Verify the popover remains open
+    assert.dom('[role="listbox"]').exists();
+  });
+
+  test('it calls `@onChange` with newly added values', async function (assert) {
+    assert.expect(3);
+
+    let selected = ['a', 'b', 'c'];
+    let options = ['a', 'b', 'c', 'd', 'e', 'f'];
+
+    let handleChange = (value: string[]) => {
+      assert.deepEqual(
+        value,
+        ['a', 'b', 'c', 'd'],
+        'Expected "d" to be added on change'
+      );
+      assert.step('handleChange');
+    };
+
+    // NOTE: starting with selected options already!
+    await render(<template>
+      <Multiselect
+        @selected={{selected}}
+        @options={{options}}
+        @onChange={{handleChange}}
+        data-multiselect
+      >
+        <:noResults>No results</:noResults>
+
+        <:remove as |remove|>
+          <remove.Remove @label="Remove" />
+        </:remove>
+
+        <:default as |multiselect|>
+          <multiselect.Option>{{multiselect.option}}</multiselect.Option>
+        </:default>
+      </Multiselect>
+    </template>);
+
+    // Select another option, "d"
+    await fillIn('[data-multiselect]', 'd');
+    await triggerKeyEvent('[data-multiselect]', 'keydown', 'Enter');
+
+    assert.verifySteps(['handleChange']);
+  });
+
+  test('it calls `@onChange` when a chip is removed by click the "X"/remove button', async function (assert) {
+    assert.expect(6);
+
+    let selected = ['a', 'b', 'c'];
+    let options = ['a', 'b', 'c'];
+
+    let handleChange = (value: string[]) => {
+      assert.deepEqual(
+        value,
+        ['a', 'c'],
+        'Expected "b" to be removed on change'
+      );
+      assert.step('handleChange');
+    };
+
+    // NOTE: starting with selected options already!
+    await render(<template>
+      <Multiselect
+        @selected={{selected}}
+        @options={{options}}
+        @onChange={{handleChange}}
+        data-multiselect
+      >
+        <:noResults>No results</:noResults>
+
+        <:remove as |remove|>
+          <remove.Remove @label="Remove" />
+        </:remove>
+
+        <:default as |multiselect|>
+          <multiselect.Option>{{multiselect.option}}</multiselect.Option>
+        </:default>
+      </Multiselect>
+    </template>);
+
+    assert.dom('[data-multiselect-remove-option]').exists({ count: 3 });
+
+    let removeButtons = document.querySelectorAll(
+      '[data-multiselect-remove-option]'
+    );
+
+    assert.ok(removeButtons, 'Expected queryable remove buttons');
+    assert.ok(
+      removeButtons[1],
```

Comment:
Does `removeButtons[1]` instanceof HTMLElement` let you remove the cast below?

Reply:
Unfortunately it looks like it's still not happy with that 🙃

Reply:
Derp. Asserting on an element of the array doesn't change its type. It makes little difference, but you'd have to assign `removeButtons[1]` to a variable, then assert that variable.

Reply:
OH! Let me try that!

Reply:
(Talked, it still doesn't work lol)